integration/msp/rsaca_test.go

```diff
@@ -0,0 +1,467 @@
+
+	certFilename := filepath.Join(dir, ca.certFilename())
+	writeCertificate(certFilename, ca.certBytes)
+
+	keyFilename := filepath.Join(dir, fmt.Sprintf("%x_sk", ca.cert.SubjectKeyId))
+	writeKey(keyFilename, ca.signer)
+}
+
+func writeCertificate(filename string, der []byte) {
+	certFile, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE, 0644)
+	Expect(err).NotTo(HaveOccurred())
+	defer certFile.Close()
+	err = pem.Encode(certFile, &pem.Block{Type: "CERTIFICATE", Bytes: der})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func writeKey(filename string, signer crypto.Signer) {
+	keyFile, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE, 0600)
+	Expect(err).NotTo(HaveOccurred())
+	defer keyFile.Close()
+	derKey, err := x509.MarshalPKCS8PrivateKey(signer)
+	Expect(err).NotTo(HaveOccurred())
+	err = pem.Encode(keyFile, &pem.Block{Type: "PRIVATE KEY", Bytes: derKey})
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func writeVerifyingMSP(mspDir string, signCA, tlsCA *CA, nodeOUs bool) {
+	for _, dir := range []string{"admincerts", "cacerts", "tlscacerts"} {
+		err := os.MkdirAll(filepath.Join(mspDir, dir), 0755)
+		Expect(err).NotTo(HaveOccurred())
+	}
+	if nodeOUs {
+		configFilename := filepath.Join(mspDir, "config.yaml")
+		writeConfigYaml(configFilename, filepath.Join("cacerts", signCA.certFilename()), nodeOUs)
+	}
+	writeCertificate(filepath.Join(mspDir, "cacerts", signCA.certFilename()), signCA.certBytes)
+	writeCertificate(filepath.Join(mspDir, "tlscacerts", tlsCA.certFilename()), tlsCA.certBytes)
+}
+
+func writeLocalMSP(baseDir, name string, signOUs, sans []string, signCA, tlsCA *CA, adminCertPem []byte, nodeOUs, client bool) {
+	mspDir := filepath.Join(baseDir, "msp")
+	err := os.MkdirAll(mspDir, 0755)
+	Expect(err).NotTo(HaveOccurred())
+	writeVerifyingMSP(mspDir, signCA, tlsCA, nodeOUs)
+
+	for _, dir := range []string{"admincerts", "keystore", "signcerts"} {
+		err := os.MkdirAll(filepath.Join(mspDir, dir), 0755)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	if !nodeOUs && len(adminCertPem) != 0 {
+		block, _ := pem.Decode(adminCertPem)
+		adminCert, err := x509.ParseCertificate(block.Bytes)
+		Expect(err).NotTo(HaveOccurred())
+		err = ioutil.WriteFile(filepath.Join(mspDir, "admincerts", certFilename(adminCert.Subject.CommonName)), adminCertPem, 0644)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// create signcert
+	priv := generateECKey()
+	signcertBytes, signcert := signCA.issueSignCertificate(name, signOUs, priv.Public())
+	signcertFilename := filepath.Join(mspDir, "signcerts", certFilename(name))
+	writeCertificate(signcertFilename, signcertBytes)
+	signcertKeyFilename := filepath.Join(mspDir, "keystore", fmt.Sprintf("%x_sk", signcert.SubjectKeyId))
+	writeKey(signcertKeyFilename, priv)
+
+	// populate tls
+	tlsDir := filepath.Join(baseDir, "tls")
+	err = os.MkdirAll(tlsDir, 0755)
+	Expect(err).NotTo(HaveOccurred())
+	writeCertificate(filepath.Join(tlsDir, "ca.crt"), tlsCA.certBytes)
+
+	tlsKey := generateRSAKey()
```

Comment:
I understand you're testing this, but why is this needed? BCCSP never processes any TLS certificates so the production code change is not affecting this part of the test.

Reply:
Nothing in the integration tests RSA TLS certificates. Since this test is explicitly targeted to RSA, it seemed reasonable to have something here to help prevent future regressions.